.github/workflows/main.yml

```diff
@@ -23,18 +23,60 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2
+
+      - name: Prepare environment variables
+        run: |
+          if [ "${{ github.repository }}" != "freeipa/freeipa-operator" ]; then
+            if [ "${{ secrets.ENV }}" == "" ]; then
+              echo "ERROR:You need to define ENV secret with your environment."
+              echo "INFO:See DEVOPS.md file for further information"
+              cat DEVOPS.md
+              exit 1
+            else
+              [ -e "bin" ] || mkdir bin
+              cat > env.ci <<EOF
+          ${{ secrets.ENV }}
+          EOF
+            fi
+          else
+            cp -vf build/ci/env env.ci
+          fi
+          [ -e "bin" ] || mkdir bin
+          [ -e "testbin" ] || mkdir testbin
+          [ -e "~/.cache/go-build" ] || mkdir -p ~/.cache/go-build
+          [ -e "~/go/pkg/mod" ] || mkdir -p ~/go/pkg/mod
+
+      # yamllint disable-line rule:line-length
+      - run: env -i bash -c 'set -a; source env.ci; env --unset=PWD --unset=SHLVL --unset=_' >> "$GITHUB_ENV"
+      - uses: actions/upload-artifact@v3
+        with:
+          name: env
+          path: env.ci
```

Comment:
Upload the `env.ci` as an artifact. This is used to share it into the different jobs, so it can be just downloaded and used to define the needed environment variables.